packages/toc/src/toc_item.tsx

```diff
@@ -2,7 +2,120 @@
 // Distributed under the terms of the Modified BSD License.
 
 import * as React from 'react';
-import { IHeading } from './utils/headings';
+import * as ts from 'typescript';
+import { CommandRegistry } from '@lumino/commands';
+import { IHeading, INotebookHeading } from './utils/headings';
+import { Menu } from '@lumino/widgets';
+import { CodeCell } from '@jupyterlab/cells';
+import { NotebookPanel } from '@jupyterlab/notebook';
+import { CommandIDs } from './toc_tree';
+
+/**
+ * Adds commands to command registry.
+ *
+ * @param commands - command registry
+ * @param toc - List of headings
+ * @param heading - heading at current position
+ * @returns
+ */
+const addCommands = (
+  commands: CommandRegistry,
+  toc: IHeading[],
+  heading: IHeading
+) => {
+  commands.addCommand(CommandIDs.runCells, {
+    execute: args => {
+      const runCells = args['runCells'] as string;
+      const func = ts.transpile(runCells);
+      const runnable = eval(func);
```

Comment:
Is this really needed? Wouldn't it be fine to just do `const runnable = NestedCodeCells`? Also, it seems to me that it still uses a closure to remember `commands`, `toc`, and `heading`. I would suggest to use args to pass those instead.